FEXCore/Source/Interface/Core/OpcodeDispatcher.h

```diff
@@ -2100,15 +2100,21 @@ class OpDispatchBuilder final : public IREmitter {
     return CurrentDeferredFlags.Type == FlagsGenerationType::TYPE_NONE;
   }
 
-  template<typename F>
-  void Calculate_ShiftVariable(Ref Shift, F&& Calculate) {
+  void ZeroShiftResult() {}
+
+  template<typename F, typename FF>
+  void Calculate_ShiftVariable(Ref Shift, F&& Calculate, FF&& ZeroShiftResult, bool NeedsZeroShiftResult = true) {
```

Comment:
Why not make ZeroShiftResult nullable, calling it if present and ignoring it if absent? Then the boolean flag goes away and the two can't go out of sync. And we also don't need the DoNothing thunk below

Reply:
nullable indeed